packages/tooltip/src/vaadin-tooltip.js

```diff
@@ -703,6 +704,19 @@ class Tooltip extends OverlayClassMixin(ThemePropertyMixin(ElementMixin(PolymerE
     }
   }
 
+  /** @private */
+  __onOverlayMouseEnter(event) {
+    if (this.manual) {
+      return;
+    }
+
+    // Retain opened state when moving pointer over the overlay.
+    // See https://github.com/vaadin/web-components/issues/6316
+    if (!this.__isTargetHidden && !this.__hoverInside) {
```

Comment:
These checks aren't currently tested

Reply:
Updated the PR to replace these checks with the actual check needed here: whether the overlay closing is in progressor not. As it can be only triggered by auto-opened tooltip, we don't need to check for `manual`.

Reply:
It still doesn't need the check for `isClosing`. Tests pass fine with `this._stateController.open({ immediate: true });` alone in the method

Reply:
Right, I was just not sure how to test this properly. Apparently I'll have to use spy on the `_stateController`.

Reply:
Ah, if it's too difficult to test, then a clarifying comment is enough